test/dpul_collections/indexing_pipeline/integration/full_integration_test.exs

```diff
@@ -71,4 +84,62 @@ defmodule DpulCollections.IndexingPipeline.FiggyFullIntegrationTest do
     transformer |> Broadway.stop(:normal)
     indexer |> Broadway.stop(:normal)
   end
+
+  def index_record_id(id) do
+    # Get record with a description.
+    record = IndexingPipeline.get_figgy_resource!(id)
+    # Write a current hydration marker right before that marker.
+    marker = IndexingPipeline.DatabaseProducer.CacheEntryMarker.from(record)
+
+    earlier_marker = %IndexingPipeline.DatabaseProducer.CacheEntryMarker{
+      id: marker.id,
+      timestamp: DateTime.add(marker.timestamp, -1, :microsecond)
+    }
+
+    IndexingPipeline.write_processor_marker(%{
+      type: IndexingPipeline.Figgy.HydrationProducerSource.processor_marker_key(),
+      cache_version: 1,
+      cache_location: earlier_marker.timestamp,
+      cache_record_id: earlier_marker.id
+    })
+
+    # Start the figgy producer
+    {:ok, indexer} = Figgy.IndexingConsumer.start_link(cache_version: 1, batch_size: 50)
+    {:ok, transformer} = Figgy.TransformationConsumer.start_link(cache_version: 1, batch_size: 50)
+
+    # Control hydration indexing.
+    {:ok, hydrator} =
+      Figgy.HydrationConsumer.start_link(
+        cache_version: 1,
+        batch_size: 50,
+        producer_module: MockFiggyHydrationProducer,
+        producer_options: {self(), 1}
+      )
+
+    # Index one.
+    MockFiggyHydrationProducer.process(1)
+
+    task =
+      Task.async(fn -> wait_for_indexed_count(1) end)
+
+    Task.await(task, 15000)
+    document = Solr.find_by_id(id)
+    {hydrator, transformer, indexer, document}
+  end
+
+  test "indexes description" do
+    {hydrator, transformer, indexer, document} =
+      index_record_id("26713a31-d615-49fd-adfc-93770b4f66b3")
+
+    assert %{"description_txtm" => [first_description | _tail]} = document
+    assert first_description |> String.starts_with?("Asra-Panahi") == true
+    # Language detection
+    assert %{"description_txtm_en" => [first_description | _tail]} = document
+    assert first_description |> String.starts_with?("Asra-Panahi") == true
+    assert %{"language_ss" => ["en"]} = document
```

Comment:
Is this language field specific to the description?

Reply:
I think if we detect languages in more fields it'll append that here